app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/provision/payload.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+# Methods to create a new json payload.
+module ManageIQ::Providers::IbmCloud::VPC::CloudManager::Provision::Payload
+  # Create the hash that will be sent to the provider for provisioning.
+  # @return [Hash] A complete hash for provisioning.
+  def prepare_for_clone_task
+    logger(__method__).debug("Final IBM VPC provision workflow form options. #{options}")
+    payload = {
+      :name                      => get_option(:vm_target_name),
+      :keys                      => [{:id => get_option(:guest_access_key_pair)}],
+      :profile                   => {:name => get_option_last(:provision_type)},
+      :image                     => {:id => vm_image[:ems_ref]},
+      :zone                      => {:name => get_option_last(:placement_availability_zone)},
+      :vpc                       => {:id => get_option(:cloud_network)},
+      :primary_network_interface => {
+        :name            => 'eth0',
+        :subnet          => {:id => get_option(:cloud_subnet)},
+        :security_groups => [{:id => get_option(:security_groups)}]
+      },
+      :volume_attachments        => cloud_volumes,
+      :boot_volume_attachment    => {
+        :volume => {
+          :name    => "#{get_option(:vm_target_name)}-boot",
+          :profile => {:name => get_option_last(:storage_type)}
+        }
+      }
+    }
+    resource_group(payload)
+  rescue => e
+    logger(__method__).log_backtrace(e)
+  end
+
+  # Add a resource group if it is selected. If not selected then VPC will use the default resource_group.
+  # @param payload [Hash] Payload for provision.
+  # @return [Hash] Payload for provision with optional resource_group set.
+  def resource_group(payload)
+    resource_group = get_option(:resource_group)
+    payload[:resource_group] = {:id => resource_group} unless resource_group.nil?
```

Comment:
If this is going to modify the argument the method should be `resource_group!`

Reply:
It's flow control not something that gets called by itself. I did have them as private methods but if I'm able to write specs on this I want to be able call these methods in isolation.